src/util/exportRequestBuilder.ts

```diff
@@ -0,0 +1,97 @@
+import { SupportedExportTypes } from '@/components/query-selector/export-type';
+import { TypeElement } from '@/state/type-element-params-state';
+
+/*
+ * Parameters to build export request string.
+ */
+export interface BuilderRequest {
+  baseUrl: string;
+  exportType: SupportedExportTypes;
+  id?: string;
+  queryParams: BuilderRequestQueryParams;
```

Comment:
```suggestion
  queryParams: BuilderRequestQueryParams;
  id?: string;
```
I think putting the optional ones at the end is good practice